### PR TITLE
Ethash

### PR DESCRIPTION
**Pull request by xcthulhu**
Along with a minor protocol change to only include just one uncle, I have updated go-ethereum to use ethash for mining.

### CODE REVIEW

File: core/transaction_pool.go [hunk 1]
```diff
@@ -117,8 +117,13 @@ func (self *TxPool) add(tx *types.Transaction) error {
 	} else {
 		to = "[NEW_CONTRACT]"
 	}
-
-	txplogger.Debugf("(t) %x => %s (%v) %x\n", tx.From()[:4], to, tx.Value, tx.Hash())
+	var from string
+	if len(tx.From()) > 0 {
+		from = ethutil.Bytes2Hex(tx.From()[:4])
+	} else {
+		from = "INVALID"
```
Comment (obscuren) [inline core/transaction_pool.go]: Addition is fine but should return an error instead

**Comment by obscuren** [comment]
Undo the Dagger PoW deletion

**Comment by obscuren** [comment]
We've changed from 1 uncle to 2 uncles per block. Just a FYI. I'll update that myself.

**Comment by xcthulhu** [comment]
Apparently this isnt' needed anymore.

**Pull request closed without merging**